### DISCUSSION TITLE
MEP modeling

### DISCUSSION THREAD

**Discussion by martinurbieta** [IfcOpenShell/IfcOpenShell#4582] [General]
I am working on a project to model a basic electrical system for residencial houses with IfcOpenShell. My objective is to genenrate a IFC file with the entites and its relations, like circuit, panels, switch in order to be able to create a branch panel.
I add and assign the system, create instances for the entities like IfcLightFixture,  IfcSwitchingDevices, IfcDistributionBoards, IfcOutlet, and so on. 
I add ports and connect them according to Source-Sink  directions.
However, when I try to evaluate the results on Revit importing the IFC file, the objects are not recognized as part of the electrical system, circuits, etc..
Maybe I am missing some relations between entities.

**Comment by aothms** [comment]
cc @CyrilWaechter any idea?

**Comment by Moult** [comment]
This is potentially a bug with Revit but we need to see the IFC to be sure. If I recall last time I tested, Revit didn't export circuits into IFC, so potentially it won't import it either.

**Comment by martinurbieta** [comment]
As I understand @Moult  is right, and customization is required in `revit File > Export > IFC > Modify Setup`.

We are working with floorplans recognition with computer vision and ML in order to model MEP system as IFC,
